optimum/onnx/transformations_utils.py

```diff
@@ -312,3 +316,223 @@ def cast_int64_tensorproto_to_int32(initializer: onnx.TensorProto, cast: bool =
 
     initializer.CopyFrom(tensor)
     initializer.name = original_name
+
+
+def _get_weights_to_tie(tied_params, torch_model: "nn.Module") -> Tuple[List[List[str]]]:
+    """
+    Find tied weights in the PyTorch model `model`, and separate them in tied weights
+    for which an untying strategy exists and do not exist.
```

Comment:
nit:
```suggestion
    for which an untying strategy exists and does not exist.
```

Reply:
Also I did not really understand the docstring.

Reply:
The implementation can only re-tie Gather and MatMul weights on the ONNX side, while there may be other tied weights on the torch side. We use the pytorch model to split the tied weights groups in one for which we have an available implementation to tie on the ONNX side, and an other for which we don't have (we will detect whether weights are duplicated, but won't fix).